research/conditional/utils/conditional_trainer.py

```diff
@@ -447,6 +451,27 @@ def _save_weights(self, step):
                     _ = self.model(
                         torch.zeros((self.batch_size, self.cutoff), dtype=torch.int)
                     )
+            save_checkpoint(
+                self.model,
+                self.optimizer,
+                self.scaler,
+                self.save_weights_path,
+                self.rank,
+                step,
+                self.logger,
+            )
+
+    def _repeater_rerun(self, step, repeater_job_end_time: Optional[int], buffer=15 * 60) -> bool:
+        if repeater_job_end_time and ((repeater_job_end_time - time())) < buffer:
+            if isinstance(self.model, FSDP):
+                # for some reason, setting the model to training mode and
+                # running a forward pass is necessary to be able to save it
+                # in FSDP. God help us. #dev can i delete this, or is it a TODO tag?
+                self.model.train()
+                with torch.no_grad():
+                    _ = self.model(
+                        torch.zeros((self.batch_size, self.cutoff), dtype=torch.int)
+                    )
```

Comment:
Extract this to separate function `initialize_model_if_fsdp` (or something better)

Reply:
added to save_checkpoint fun